src/main/java/gyro/google/kms/KeyRingResource.java

```diff
@@ -0,0 +1,140 @@
+package gyro.google.kms;
+
+import java.util.Set;
+
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.cloud.kms.v1.KeyManagementServiceClient;
+import com.google.cloud.kms.v1.KeyRing;
+import com.google.cloud.kms.v1.LocationName;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.scope.State;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+import gyro.google.GoogleResource;
+import gyro.google.util.Utils;
+
+/**
+ * Create a key ring.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    google::key-ring example-key-ring
+ *        location: "global"
+ *        name: "example-key-ring"
+ *    end
+ */
+@Type("key-ring")
+public class KeyRingResource extends GoogleResource implements Copyable<KeyRing> {
+
+    private String location;
+    private String name;
+
+    // Read-only
+    private String id;
+
+    /**
+     * The location of the key ring. The valid values are ``asia-east1`` or ``asia-east2`` or ``asia-northeast1`` or ``asia-northeast2`` or ``asia-northeast3`` or ``asia-south1`` or ``asia-southeast1`` or ``australia-southeast1`` or ``europe-north1`` or ``europe-west1`` or ``europe-west2`` or ``europe-west3`` or ``europe-west4`` or ``europe-west6`` or ``northamerica-northeast1`` or ``us-central1`` or ``us-east1`` or ``us-east4`` or ``us-west1`` or ``us-west2`` or ``us-west3`` or ``southamerica-east1`` or ``eur4`` or ``nam4`` or ``global`` or ``asia`` or ``europe`` or ``us``. (Required)
```

Comment:
I would say remove the valid values from the docs as well, as even that needs to be updated when a value changes